ccavenue/ccavenue.php

```diff
@@ -1,103 +1,198 @@
 <?php
 /**
- *  @copyright  Copyright (c) 2009-2013 TechJoomla. All rights reserved.
- *  @license    GNU General Public License version 2, or later
+ * @package    Common_Code
+ * @author     TechJoomla <extensions@techjoomla.com>
+ * @website    http://techjoomla.com
+ * @copyright  Copyright (C) 2005 - 2014 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-//error_reporting(E_ALL);
-//ini_set('display_errors','On');
-
-// no direct access
-defined( '_JEXEC' ) or die( 'Restricted access' );
-jimport( 'joomla.plugin.plugin' );
-require_once(dirname(__FILE__) . '/ccavenue/helper.php');
-$lang =  JFactory::getLanguage();
+// No direct access
+defined('_JEXEC') or die('Restricted access');
+jimport('joomla.plugin.plugin');
+require_once dirname(__FILE__) . '/ccavenue/helper.php';
+require_once dirname(__FILE__) . '/ccavenue/Crypto.php';
+$lang = JFactory::getLanguage();
 $lang->load('plg_payment_ccavenue', JPATH_ADMINISTRATOR);
-class  plgPaymentCcavenue extends JPlugin
-{
 
-	function __construct(& $subject, $config)
+/**
+ * plgPaymentCcavenue plugin class.
+ *
+ * @package  JGive
+ * @since    1.8
+ */
+class  PlgPaymentCcavenue extends JPlugin
+{
+	/**
+	 * Method _construct
+	 *
+	 * @param   String  &$subject  Subject
+	 * @param   String  $config    Config
+	 *
+	 * @since    1.8.1
+	 */
+	public function __construct(& $subject, $config)
 	{
 		parent::__construct($subject, $config);
-		//Set the language in the class
-		$config = JFactory::getConfig();
 
+		// Set the language in the class
+		$config = JFactory::getConfig();
 
-		//Define Payment Status codes in payu  And Respective Alias in Framework
-		$this->responseStatus= array( 'Y'=>'C', 'B'=>'P', 'N'=>'D','ERROR'  => 'E' );
+		// Define Payment Status codes in payu  And Respective Alias in Framework
+		$this->responseStatus = array('Success' => 'C', 'Failure' => 'P', 'Aborted' => 'E','ERROR' => 'E');
 	}
 
-	function buildLayoutPath($layout) {
+	/**
+	 * Method to take layout and return the file
+	 *
+	 * @param   String  $layout  Layout
+	 *
+	 * @return  file
+	 *
+	 * @since   1.8.1
+	 */
+	public function buildLayoutPath($layout)
+	{
 		$app = JFactory::getApplication();
-	if(empty($layout))
-		$layout = "default";
 
-		$core_file 	= dirname(__FILE__) . '/' . $this->_name . '/' . 'tmpl' . '/' . $layout.'.php';
-		$override		= JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' . $this->_type . '/' . $this->_name . '/' . $layout.'.php';
-		if(JFile::exists($override))
+		if (empty($layout))
+		{
+			$layout = "default";
+		}
+
+		$core_file = dirname(__FILE__) . '/' . $this->_name . '/' . 'tmpl' . '/' . $layout . '.php';
+		$override = JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' .
```

Comment:
can you change it  from 
`JPATH_BASE . '/' . 'templates' . '/' . $app->getTemplate() . '/html/plugins/' .`
=>
to 
`JPATH_BASE . '/templates/' . $app->getTemplate() . '/html/plugins/' .`